### DISCUSSION TITLE
Preserve original _atom_site_label while Structure.from_file and cifparser

### DISCUSSION THREAD

**Discussion by Ardalanhayatifar** [materialsproject/pymatgen#2503] [Q&A] [answered]
When importing a .cif file, it can contain info about different types of elements that can come handy for several reasons. For example:
```
loop_
_atom_site_label
_atom_site_fract_x
_atom_site_fract_y
_atom_site_fract_z
_atom_site_occupancy
_atom_site_U_iso_or_equiv
Fe1   0.16950   0.83040   0.63650   1.00000   0.01100
Fe2   0.33333   0.66667   0.33790   0.97000   0.01100
Fe3   0.33333   0.66667   0.95950   0.96000   0.01100
O-H1   0.00000   0.00000   0.04460   1.00000   0.00700
O2   0.33333   0.66667   0.76340   1.00000   0.00700
O3   0.16970   0.83020   0.24670   1.00000   0.00700
O4   0.52270   0.47730   0.97960   1.00000   0.00700
```
Importing using ```Structure.from_file``` ignores the types and works based on a [0-index] counter. I tried to edit my ```cif.py``.
https://github.com/materialsproject/pymatgen/blob/d8a6c479cfc5828288e13015526a780306f23f15/pymatgen/io/cif.py#L420

Is there any way that I can read the ```.cif``` and get results like this:
```
PeriodicSite: Fe1 (-1.4565, 4.2631, 5.8087) [0.1695, 0.8304, 0.6365]
PeriodicSite: Fe1 (2.9643, 3.3929, 1.2457) [0.8305, 0.6609, 0.1365]
PeriodicSite: Fe2 (1.5078, 0.8702, 1.2457) [0.3391, 0.1695, 0.1365]
PeriodicSite: Fe3 (0.0003, 1.7409, 5.8087) [0.1696, 0.3391, 0.6365]
PeriodicSite: Fe4 (4.4205, 0.8707, 1.2457) [0.8305, 0.1696, 0.1365]
PeriodicSite: Fe5 (1.4562, 4.2636, 5.8087) [0.6609, 0.8305, 0.6365]
PeriodicSite: Fe:0.960 (0.0000, 3.4225, 8.7564) [0.3333, 0.6667, 0.9595]
PeriodicSite: Fe:0.960 (2.9640, 1.7113, 4.1934) [0.6667, 0.3333, 0.4595]
PeriodicSite: Fe:0.970 (0.0000, 3.4225, 3.0837) [0.3333, 0.6667, 0.3379]
PeriodicSite: Fe:0.970 (2.9640, 1.7113, 7.6467) [0.6667, 0.3333, 0.8379]
PeriodicSite: O-H1 (0.0000, 0.0000, 0.4070) [0.0000, 0.0000, 0.0446]
PeriodicSite: O2 (0.0000, 0.0000, 4.9700) [0.0000, 0.0000, 0.5446]
PeriodicSite: O3 (0.0000, 3.4225, 6.9668) [0.3333, 0.6667, 0.7634]
PeriodicSite: O4 (2.9640, 1.7113, 2.4038) [0.6667, 0.3333, 0.2634]
```

**Comment by shyuep** [comment, marked as answer]
I have no idea what you mean by different types of elements. 0 based indexing is the recommended in programming. There are ways to group the sites by symmetry equivalent.

In general, I wouldn't recommend editing the cif.py itself to do this.

**Reply by Anny-tech** [reply]
I think, what was meant by different element type was: keeping 'Fe1' as Fe1 and not converting it to 'Fe' (just an example) in the '_atom_site_label' after parsing into a structure in pymatgen. This might come especially handy if someone is handling defects in structures. Personally I am also facing the same issue, due to which I had to use another program to get my structures. Would be great if this gets sorted.

**Reply by shyuep** [reply]
I think it is dangerous to use ext site labels for defects. You can easily run a symmetry analysis in pymatgen to get symmetrically distinct sites and put defects in the distinct sites.
